src/devices/Arduino/ArduinoBoard.cs

```diff
@@ -0,0 +1,413 @@
+        /// </summary>
+        public static List<int> CommonBaudRates()
+        {
+            return new List<int>()
+            {
+                9600,
+                19200,
+                18400,
+                57600,
+                115200,
+                230400,
+                250000,
+                500000,
+                1000000,
+                2000000,
+            };
+        }
+
+        /// <summary>
+        /// Initialize the board connection. This must be called before any other methods of this class.
+        /// </summary>
+        /// <exception cref="NotSupportedException">The Firmata firmware on the connected board is too old.</exception>
+        public virtual void Initialize()
+        {
+            if (_firmata != null)
+            {
+                throw new InvalidOperationException("Board already initialized");
+            }
+
+            _firmata = new FirmataDevice();
+            _firmata.Open(_dataStream);
+            _firmata.OnError += FirmataOnError;
+            FirmataVersion = _firmata.QueryFirmataVersion();
+            if (FirmataVersion < _firmata.QuerySupportedFirmataVersion())
+            {
+                throw new NotSupportedException($"Firmata version on board is {FirmataVersion}. Expected {_firmata.QuerySupportedFirmataVersion()}. They must be equal.");
+            }
+
+            Log($"Firmata version on board is {FirmataVersion}.");
+
+            FirmwareVersion = _firmata.QueryFirmwareVersion(out var firmwareName);
+            FirmwareName = firmwareName;
+
+            Log($"Firmware version on board is {FirmwareVersion}");
+
+            _firmata.QueryCapabilities();
+
+            _supportedPinConfigurations = _firmata.PinConfigurations; // Clone reference
+
+            Log("Device capabilities: ");
+            foreach (var pin in _supportedPinConfigurations)
+            {
+                Log(pin.ToString());
+            }
+
+            _firmata.EnableDigitalReporting();
+        }
+
+        /// <summary>
+        /// Firmware version on the device
+        /// </summary>
+        public Version FirmwareVersion
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// Name of the firmware.
+        /// </summary>
+        public string FirmwareName
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// Firmata version found on the board.
+        /// </summary>
+        public Version FirmataVersion
+        {
+            get;
+            private set;
+        }
+
+        internal FirmataDevice Firmata
+        {
+            get
+            {
+                return _firmata ?? throw new ObjectDisposedException(nameof(ArduinoBoard));
+            }
+        }
+
+        /// <summary>
+        /// Returns the list of capabilities per pin
+        /// </summary>
+        public ReadOnlyCollection<SupportedPinConfiguration> SupportedPinConfigurations
```

Comment:
NIT: it is usually better practice to return interfaces when appropriate, for example here you could do `IReadOnlyList<SupportedPinConfiguration>` and above where you are returning BaudRates you can use `IEnumerable<int>` or `IList<int>`